operations/commit_queue.go

```diff
@@ -117,3 +178,124 @@ func deleteCommitQueueItem(ctx context.Context, client client.Communicator, proj
 
 	return nil
 }
+
+type mergeParams struct {
+	projectID   string
+	ref         string
+	id          string
+	pause       bool
+	description string
+	skipConfirm bool
+	large       bool
+}
+
+func (p *mergeParams) mergeBranch(ctx context.Context, confPath string) error {
+	conf, err := NewClientSettings(confPath)
+	if err != nil {
+		return errors.Wrap(err, "problem loading configuration")
+	}
+
+	ac, _, err := conf.getLegacyClients()
+	if err != nil {
+		return errors.Wrap(err, "problem accessing evergreen service")
+	}
+
+	client := conf.GetRestCommunicator(ctx)
+	if err != nil {
+		return errors.Wrap(err, "problem accessing evergreen service")
+	}
+
+	if p.id == "" {
+		if p.projectID == "" {
+			p.projectID = conf.FindDefaultProject()
+		} else {
+			if conf.FindDefaultProject() == "" &&
+				!p.skipConfirm && confirm(fmt.Sprintf("Make %v your default project?", p.projectID), true) {
+				conf.SetDefaultProject(p.projectID)
+				if err := conf.Write(""); err != nil {
+					grip.Warningf("warning - failed to set default project: %v\n", err)
+				}
+			}
+		}
+		if p.projectID == "" {
+			return errors.New("no project identifier provided")
+		}
+		projectRef, err := ac.GetProjectRef(p.projectID)
+		if err != nil {
+			return err
+		}
+		diffData, err := getFeaturePatchInfo(projectRef.Branch, p.ref)
+		if err != nil {
+			return errors.Wrap(err, "can't generate patches")
+		}
+
+		if p.description == "" && !p.skipConfirm {
+			p.description = prompt("Enter a description for this patch (optional):")
+		}
+		if p.description == "" {
+			p.description, err = gitCmd("rev-parse", "--abbrev-ref", p.ref)
```

Comment:
if no description is provided we use the branch name, which is not equal to ref in the case of HEAD.